diff --git a/llama.cpp b/llama.cpp
index f78594a6..50a9a0ec 100644
--- a/llama.cpp
+++ b/llama.cpp
@@ -14560,7 +14560,7 @@ static std::vector<llama_vocab::id> llama_tokenize_internal(const llama_vocab &
 
                         if (vocab.tokenizer_add_space_prefix) {
                             if (!output.size() || is_prev_special) {  // prefix with space if first token
-                                raw_text = " " + raw_text;
+                                //raw_text = " " + raw_text;
                             }
                         }
 
@@ -18299,6 +18299,8 @@ size_t llama_state_set_data(struct llama_context * ctx, const uint8_t * src) {
             memcpy(ctx->logits, inp, logits_size * sizeof(float));
             inp += logits_size * sizeof(float);
         }
+        
+        ctx->logits_size = logits_size;
     }

